backbone/src/main/java/org/researchstack/backbone/utils/ObjectUtils.java

```diff
@@ -0,0 +1,39 @@
+package org.researchstack.backbone.utils;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * Created by TheMDP on 12/29/16.
+ */
+
+public class ObjectUtils {
+
+    /*
+     * Performs a deep copy on the object of type using Gson
+     * Object type must have a default constructor to work properly
+     *
+     * NOTE: this does not work with Polymorphism
+     * Use GsonSerializablePolymorphism class instead
+     */
+    public static <T> T deepCopy(Object object, Class<T> type) {
+        return deepCopy(object, type, new Gson());
+    }
+
+    /*
+    * Performs a deep copy on the object of type using Gson
+    * Object type must have a default constructor to work properly
+    *
+    * NOTE: this does not work with Polymorphism
+    * Use GsonSerializablePolymorphism class instead
+    */
+    public static <T> T deepCopy(Object object, Class<T> type, Gson gson) {
+        try {
+            String copyJson = gson.toJson(object, type);
+            return gson.fromJson(copyJson, type);
```

Comment:
In deepCopy and `DataPair` and` Data`, do we want to use java.lang.reflect.Type instead of `Class<T>`?

I think by using `Class<T>` we lose generic type information, so we should use this method instead
https://google.github.io/gson/apidocs/com/google/gson/Gson.html#fromJson-java.lang.String-java.lang.reflect.Type-